core/src/main/java/com/github/phantazmnetwork/core/gui/Gui.java

```diff
@@ -0,0 +1,287 @@
+package com.github.phantazmnetwork.core.gui;
+
+import com.github.phantazmnetwork.commons.Tickable;
+import net.kyori.adventure.text.Component;
+import net.minestom.server.entity.Player;
+import net.minestom.server.inventory.Inventory;
+import net.minestom.server.inventory.InventoryType;
+import net.minestom.server.item.ItemStack;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+import java.util.function.BiPredicate;
+
+/**
+ * Extension of {@link Inventory} designed to ease the creation of graphical user interfaces. May or may not be
+ * "dynamic". Dynamic GUIs support animations and tick all of their constituent {@link GuiItem}s.
+ */
+@SuppressWarnings("SynchronizationOnLocalVariableOrMethodParameter")
```

Comment:
why

Reply:
intelliJ doesn't like that i'm synchronizing on the GuiItems